api/src/main/java/marquez/db/OpenLineageDao.java

```diff
@@ -135,6 +145,80 @@ default UpdateLineageRow updateMarquezModel(LineageEvent event, ObjectMapper map
     return updateLineageRow;
   }
 
+  default UpdateLineageRow updateMarquezModel(DatasetEvent event, ObjectMapper mapper) {
```

Comment:
Minor: There's a far amount of duplication between `updateMarquezModel(DatasetEvent)` and `updateMarquezModel(LineageEvent)`; more of an observation than anything else. We should consider planning future refactoring work. But, for now, this implementation is totally reasonable.

Reply:
Yes, so my thought was that trying to make it generic for dataset, job and run event will make it shorter, but less readable. So this is the reason behind duplication. Also, I didn't want to modify existing method without any discussion on that. 

I would go for some improvements, like:
 1. create value class `OpenLineageDao.Daos` with all the daos required and created on demand. For example, `upsertLineageDataset` takes 8 daos arguments. 
 2. Introducing builders for methods like `upsertJob` would reduce already existing code duplication: calling method twice with 9 same params and one different. Same is happening `runDao.upsert`
 3. Introducing methods `insertInputFacets` and `insertOutputFacets`

I don't mind working on those in this PR as along as it is still readable for the reviewer. Let me know what do you think?

Reply:
Hmm, we can hold off on the refactoring. Personally, we should look into using a jdbi [`handler`](https://jdbi.org/#_obtaining_a_managed_handle) and not restrict ourselves to using DAOs (which really are great for simpler queries, but less ideal for more complex ones).